polkadot/runtime/parachains/src/initializer.rs

```diff
@@ -328,6 +332,11 @@ impl<T: Config> Pallet<T> {
 	{
 		Self::on_new_session(changed, session_index, validators, queued)
 	}
+
+	/// Return whether at the end of this block a new session will be initialized.
+	pub(crate) fn upcoming_session_change() -> bool {
+		!BufferedSessionChanges::<T>::get().is_empty()
```

Comment:
The documentation for this says:
`Buffered session changes along with the block number at which they should be applied` , should we actually look at the block number ?

Reply:
The documentation is wrong. `BufferedSessionChanges` stores only: the session index, the new validators and the upcoming validators.
`BufferedSessionChanges` is only being set at the beginning of the block where a session change will occur. The change is buffered and applied in on_finalize. So this storage item is only set for the duration of the one block where the session change occurs